src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -88,14 +88,30 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
 
 
 class BasicCrawler(Generic[TCrawlingContext]):
-    """Provides a simple framework for parallel crawling of web pages.
-
-    The URLs to crawl are fed either from a static list of URLs or from a dynamic queue of URLs enabling recursive
-    crawling of websites.
-
-    `BasicCrawler` is a low-level tool that requires the user to implement the page download and data extraction
-    functionality themselves. If we want a crawler that already facilitates this functionality, we should consider using
-    one of its subclasses.
+    """A versatile web crawler for parallel URL fetching.
+
+    Provides a basic framework for crawling websites, allowing customization of request handling and data extraction.
```

Comment:
The class docstring should be much longer in this case. Checkout the JS version for inspiration [BasicCrawler]( https://crawlee.dev/api/basic-crawler/class/BasicCrawler). You should describe the features that are provided by the `BasicCrawler` - scaling, request routing, sessions, proxies, retries, statistics ...